frontend/public/components/app.jsx

```diff
@@ -132,13 +134,66 @@ const App = (props) => {
     }
   }, []);
 
+  const onCSPViolation = React.useCallback(
+    (event) => {
+      // eslint-disable-next-line no-console
+      console.warn('Content-Security-Policy: violation detected', event);
+
+      // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent
+      const cspReportObject = _.pick(event, [
+        // The URI of the resource that was blocked because it violates a policy.
+        'blockedURI',
+        // The column number in the document or worker at which the violation occurred.
+        'columnNumber',
+        // Whether the user agent is configured to enforce or just report the policy violation.
+        'disposition',
+        // The URI of the document or worker in which the violation occurred.
+        'documentURI',
+        // The directive that was violated.
+        'effectiveDirective',
+        // The line number in the document or worker at which the violation occurred.
+        'lineNumber',
+        // The policy whose enforcement caused the violation.
+        'originalPolicy',
+        // The URL for the referrer of the resources whose policy was violated, or null.
+        'referrer',
+        // A sample of the resource that caused the violation, usually the first 40 characters.
+        // This will only be populated if the resource is an inline script, event handler or style.
+        'sample',
+        // If the violation occurred as a result of a script, this will be the URL of the script.
+        'sourceFile',
+        // HTTP status code of the document or worker in which the violation occurred.
+        'statusCode',
+      ]);
+
+      // Attempt to infer Console plugin name from the CSP report object
+      const pluginAssetBaseURL = `${document.baseURI}api/plugins/`;
+      const pluginName = cspReportObject.sourceFile?.startsWith(pluginAssetBaseURL)
+        ? cspReportObject.sourceFile.substring(pluginAssetBaseURL.length).split('/')[0]
+        : null;
+
+      fireTelemetryEvent('Content-Security-Policy violation', {
```

Comment:
@vojtechszocs based on this the reported violation will have following format:
`"Content-Security-Policy violation | <cspReportObject> | <pluginName>"`

I assume that `cspReportObject` should contain details about the violation itself.

@luciddreamz could you please chime in here if this is a proper format, or if there is a better one. Thank you.

Reply:
@jhadvig @luciddreamz Hi, CSP violation data sent from Console to telemetry should have the following format.

```ts
{

  // --- Standard CSP violation data ---

  // The URI of the resource that was blocked because it violates a policy.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/blockedURI
  blockedURI: string,
  // The column number in the document or worker at which the violation occurred.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/columnNumber
  columnNumber: number | null,
  // Whether the user agent is configured to enforce or just report the policy violation.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/disposition
  disposition: "enforce" | "report",
  // The URI of the document or worker in which the violation occurred.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/documentURI
  documentURI: string,
  // The directive that was violated.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/effectiveDirective
  effectiveDirective: string,
  // The line number in the document or worker at which the violation occurred.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/lineNumber
  lineNumber: number | null,
  // The policy whose enforcement caused the violation.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/originalPolicy
  originalPolicy: string,
  // The URL for the referrer of the resources whose policy was violated, or null.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/referrer
  referrer: string | null,
  // A sample of the resource that caused the violation, usually the first 40 characters.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/sample
  sample: string | null,
  // If the violation occurred as a result of a script, this will be the URL of the script.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/sourceFile
  sourceFile: string | null,
  // HTTP status code of the document or worker in which the violation occurred.
  // https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent/statusCode
  statusCode: number,

  // --- OCP Console specific data ---

  // Name of the Console dynamic plugin that caused the violation.
  // Null value means the violation seems to originate from the Console application itself.
  pluginName: string | null,

}
```

Reply:
@jhadvig @luciddreamz Sample CSP violation data instance:

```js
{
  "blockedURI": "eval",
  "columnNumber": 8,
  "disposition": "report",
  "documentURI": "http://localhost:9000/dashboards",
  "effectiveDirective": "script-src",
  "lineNumber": 208990,
  "originalPolicy": "default-src 'self' http://localhost:8080 ws://localhost:8080; base-uri 'self' http://localhost:8080 ws://localhost:8080; img-src 'self' http://localhost:8080 ws://localhost:8080 data:; font-src 'self' http://localhost:8080 ws://localhost:8080 data:; script-src 'self' http://localhost:8080 ws://localhost:8080; style-src 'self' http://localhost:8080 ws://localhost:8080 'unsafe-inline'; frame-src 'none'; frame-ancestors 'none'; object-src 'none'",
  "referrer": "",
  "sample": "",
  "sourceFile": "http://localhost:9000/static/vendors~main-8601b679a80dba767713.js",
  "statusCode": 200,
  "pluginName": null,
}
```

It seems that Firefox uses empty strings instead of null values, but that can be sanitized as a follow-up.